numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -42,6 +42,32 @@
 #include "npy_cblas.h"
 #include "npy_buffer.h"
 
+
+/*
+ * Define a stack allocated dummy array with only the minimum information set:
+ *   1. The descr, the main field interesting here.
+ *   2. The flags, which are needed for alignment;.
+ *   3. The type is set to NULL and the base is the original array, if this
+ *      is used within a subarray getitem to create a new view, the base
+ *      must be walked until the type is not NULL.
+ *
+ * The following should create errors in debug mode (if deallocated
+ * incorrectly), since base would be incorrectly decref'd as well.
+ * This is especially important for nonzero and copyswap, which may run with
+ * the GIL released.
```

Comment:
If this singleton can be used with the GIL released, it can be modified by other threads, no?

Reply:
No, it is stack local to the function call using it.

Reply:
Ahh, right, not `static PyArrayObject_fields`, just `PyArrayObject_fields`.